core/test/base/array.cpp

```diff
@@ -304,16 +315,168 @@ TYPED_TEST(Array, ChangesExecutors)
+    EXPECT_EQ(array.get_data()[2], TypeParam{3});
+    EXPECT_EQ(array.get_data()[3], TypeParam{4});
+    EXPECT_EQ(array.get_num_elems(), 4);
+    ASSERT_EQ(view.get_num_elems(), 4);
+}
+
+
+TYPED_TEST(Array, CopyArrayToView)
+{
+    TypeParam data[] = {1, 2, 3};
+    auto view = gko::Array<TypeParam>::view(this->exec, 3, data);
+    gko::Array<TypeParam> array_size2(this->exec, {5, 4});
+    gko::Array<TypeParam> array_size4(this->exec, {5, 4, 2, 1});
+
+    view = array_size2;
+
+    EXPECT_EQ(data[0], TypeParam{5});
+    EXPECT_EQ(data[1], TypeParam{4});
+    EXPECT_EQ(data[2], TypeParam{3});
+    EXPECT_EQ(view.get_num_elems(), 3);
+    EXPECT_EQ(array_size2.get_num_elems(), 2);
+    ASSERT_THROW(view = array_size4, gko::OutOfBoundsError);
+}
+
+
+TYPED_TEST(Array, MoveArrayToArray)
+{
+    gko::Array<TypeParam> array(this->exec, {1, 2, 3});
+    gko::Array<TypeParam> array2(this->exec, {5, 4, 2, 1});
+
+    array = std::move(array2);
+
+    EXPECT_EQ(array.get_data()[0], TypeParam{5});
+    EXPECT_EQ(array.get_data()[1], TypeParam{4});
+    EXPECT_EQ(array.get_data()[2], TypeParam{2});
+    EXPECT_EQ(array.get_data()[3], TypeParam{1});
+    EXPECT_EQ(array.get_num_elems(), 4);
+    EXPECT_EQ(array2.get_data(), nullptr);
+    ASSERT_EQ(array2.get_num_elems(), 0);
+}
+
+
+TYPED_TEST(Array, MoveViewToView)
+{
+    TypeParam data[] = {1, 2, 3, 4};
+    auto view = gko::Array<TypeParam>::view(this->exec, 4, data);
+    TypeParam data2[] = {5, 4, 2};
+    auto view2 = gko::Array<TypeParam>::view(this->exec, 3, data2);
+
+    view = std::move(view2);
+
+    EXPECT_EQ(view.get_data(), data2);
+    EXPECT_EQ(view.get_data()[0], TypeParam{5});
+    EXPECT_EQ(view.get_data()[1], TypeParam{4});
+    EXPECT_EQ(view.get_data()[2], TypeParam{2});
+    EXPECT_EQ(view.get_num_elems(), 3);
+    EXPECT_EQ(view2.get_data(), nullptr);
+    ASSERT_EQ(view2.get_num_elems(), 0);
+}
+
+
+TYPED_TEST(Array, MoveViewToArray)
+{
+    TypeParam data[] = {1, 2, 3, 4};
+    gko::Array<TypeParam> array(this->exec, {5, 4, 2});
+    auto view = gko::Array<TypeParam>::view(this->exec, 4, data);
+
+    array = std::move(view);
+
+    EXPECT_EQ(array.get_data(), data);
+    EXPECT_EQ(array.get_data()[0], TypeParam{1});
+    EXPECT_EQ(array.get_data()[1], TypeParam{2});
+    EXPECT_EQ(array.get_data()[2], TypeParam{3});
+    EXPECT_EQ(array.get_data()[3], TypeParam{4});
+    EXPECT_EQ(array.get_num_elems(), 4);
+    EXPECT_EQ(data[0], TypeParam{1});
+    EXPECT_EQ(data[1], TypeParam{2});
+    EXPECT_EQ(data[2], TypeParam{3});
+    EXPECT_EQ(data[3], TypeParam{4});
+    EXPECT_EQ(view.get_data(), nullptr);
+    ASSERT_EQ(view.get_num_elems(), 0);
+}
+
+
+TYPED_TEST(Array, MoveArrayToView)
+{
+    TypeParam data[] = {1, 2, 3};
+    auto view = gko::Array<TypeParam>::view(this->exec, 3, data);
+    gko::Array<TypeParam> array_size2(this->exec, {5, 4});
+    gko::Array<TypeParam> array_size4(this->exec, {5, 4, 2, 1});
+
+    view = std::move(array_size2);
```

Comment:
Maybe check the type of `view` ?

Reply:
You would need to access the `data_` internal storage to check its deleter type, which we cannot with our interface and I'm not sure we want this to happen. So this is a bit tricky.

Reply:
Is is not possible to create a dummy array class in the test file and test locally ?